src/main/java/killua/ui/Ui.java

```diff
@@ -0,0 +1,209 @@
+package killua.ui;
+
+import killua.task.Task;
+import killua.util.TaskList;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+/**
+ * Represents the user interface of the Killua task manager.
+ * It handles the display of messages and reading of user input.
+ */
+public class Ui {
+
+    private final Scanner in;
+    private static final String LINE = "______________________________________________________________________________";
+
+    /**
+     * Constructs a Ui instance with default input and output streams.
+     */
+    public Ui() {
+        this(System.in, System.out);
+    }
+
+    /**
+     * Constructs a Ui instance with specified input and output streams.
+     *
+     * @param in The input stream to read user input from.
+     * @param out The output stream to print messages to.
+     */
+    public Ui(InputStream in, PrintStream out) {
+        this.in = new Scanner(in);
+    }
+
+    /**
+     * Displays a welcome message and a list of available commands.
+     */
+    public void showWelcomeMessage() {
+        printLine();
+        System.out.println("Welcome to Killua Task Manager!");
+        System.out.println("Here are some commands you can use:");
+        System.out.println("  bye - Exit the application");
+        System.out.println("  list - List all tasks");
+        System.out.println("  mark <task number> - Mark a task as done");
+        System.out.println("  unmark <task number> - Mark a task as not done yet");
+        System.out.println("  delete <task number> - Delete a task");
+        System.out.println("  todo <description> - Add a new todo task");
+        System.out.println("  deadline <description> /by <yyyy-mm-dd> OR ");
+        System.out.println("  deadline <description> /by <yyyy-mm-dd hh:mm> - Add a new deadline task");
+        System.out.println("  event <description> /from <yyyy-mm-dd> /to <yyyy-mm-dd> OR ");
+        System.out.println("  event <description> /from <yyyy-mm-dd hh:mm> /to <yyyy-mm-dd hh:mm> - Add a new event task");
+        printLine();
+    }
+
+    /**
+     * Prints a separator line for formatting purposes.
+     */
+    public void printLine() {
```

Comment:
I like how you have a `printLine` method to abstract out the printing of a line and makes your code cleaning. Should you do that for general printint too? So you would not have to keep using `System.out.println`.